pkg/kubelet/cm/topologymanager/scope_test.go

```diff
@@ -100,17 +103,23 @@ func TestRemoveContainer(t *testing.T) {
 		},
 	}
 	var len1, len2 int
+	var lenHints1, lenHints2 int
 	scope := scope{}
-	scope.podMap = make(map[string]string)
+	scope.podMap = containermap.NewContainerMap()
+	scope.podTopologyHints = podTopologyHints{}
 	for _, tc := range testCases {
-		scope.podMap[tc.containerID] = string(tc.podUID)
+		scope.podMap.Add(string(tc.podUID), tc.name, tc.containerID)
+		scope.podTopologyHints[string(tc.podUID)] = make(map[string]TopologyHint)
+		scope.podTopologyHints[string(tc.podUID)][tc.name] = TopologyHint{}
 		len1 = len(scope.podMap)
+		lenHints1 = len(scope.podTopologyHints)
 		err := scope.RemoveContainer(tc.containerID)
 		len2 = len(scope.podMap)
+		lenHints2 = len(scope.podTopologyHints)
 		if err != nil {
 			t.Errorf("Expected error to be nil but got: %v", err)
 		}
-		if len1-len2 != 1 {
+		if len1-len2 != 1 || lenHints1-lenHints2 != 1 {
 			t.Errorf("Remove Pod resulted in error")
 		}
```

Comment:
Can we break this into two checks with different errors based on which one was incorrect?

Reply:
Yeah, you are right, I will update it